### PR TITLE
Bookstore Project: Reducers and Actions

### PR DESCRIPTION
**Pull request by theodoku**
**1. Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).**
**2. Create a directory to contain all  Redux logic** 
**3. Define a slice of state for books that Includes an array of books**
**4. Include a reducer that adds a book and Include a reducer that removes a book**
**5. Define a slice of state for categories that Includes an array of categories**

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        books: state.books.filter((book) => book.id !== action.payload),
+      };
+    default:
+      return state;
+  }
+};
+
+export {
+  ADD_BOOK_ITEM, REMOVE_BOOK_ITEM, addBookItem, removeBookItem, booksReducer,
+};
```
Comment (TedLivist) [inline src/redux/books/booksSlice.js]: - [x] Your implementation of the book reducer and actions do not follow the requirements. You are required to import `createSlice` from the redux toolkit and use it to set up your reducer for the book. Kindly refactor this file to fit this requirement. You can refer to [this documentation](https://redux-toolkit.js.org/api/createSlice).

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      return {
+        ...state,
+        categories: action.payload,
+      };
+    default:
+      return state;
+  }
+};
+
+export { SET_CATEGORIES, setCategories, categoriesReducer };
```
Comment (TedLivist) [inline src/redux/categories/categories.js]: - [ ] Your implementation of the category reducer and actions do not follow the requirements. You are required to import `createSlice` from the redux toolkit and use it to set up your reducer for the category. Kindly refactor this file to fit this requirement. You can refer to [this documentation](https://redux-toolkit.js.org/api/createSlice).

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @theodoku,

Good job so far!
✔️ Use of `@reduxjs/toolkit` to configure store
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (theodoku) [reply src/redux/books/booksSlice.js]: booksSlice.js file refactored to import createSlice from redux toolkit✔

File: src/redux/categories/categories.js (hunk 2 above)
Comment (theodoku) [reply src/redux/categories/categories.js]: categories.js file refactored to import createSlice from redux toolkit✔

**Review by osamaashraf6 — APPROVED** [review]
Hi @theodoku 😊, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
MinionsGIF

## Highlights🎉 🎉 
- No linter errors present
- The PR has a proper title and a descriptive body 👍

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**